pkg/controller/podgc/gc_controller.go

```diff
@@ -308,7 +316,30 @@ func (o byCreationTimestamp) Less(i, j int) bool {
 	return o[i].CreationTimestamp.Before(&o[j].CreationTimestamp)
 }
 
-func (gcc *PodGCController) deletePod(ctx context.Context, namespace, name string) error {
-	klog.InfoS("PodGC is force deleting Pod", "pod", klog.KRef(namespace, name))
-	return gcc.kubeClient.CoreV1().Pods(namespace).Delete(ctx, name, *metav1.NewDeleteOptions(0))
+func (gcc *PodGCController) markFailedAndDeletePod(ctx context.Context, pod *v1.Pod) error {
+	return gcc.markFailedAndDeletePodWithCondition(ctx, pod, nil)
+}
+
+func (gcc *PodGCController) markFailedAndDeletePodWithCondition(ctx context.Context, pod *v1.Pod, condition *v1.PodCondition) error {
```

Comment:
this function and  addConditionAndDeletePod could be combined into a single one and save some repetition and tests

Reply:
Do you mean that instead of introducing the proxy function `markFailedAndDeletePod` we could just call `markFailedAndDeletePodWithCondition` with nil? Potentially, this was the initial implementation, but it was suggested in a previous review comment to introduce the proxy function to improve readability. I think I agree with this opinion, passing nil from multiple places often looks awkward. I don't think it would affect the set of testing scenarios.